pallets/portfolio/src/lib.rs

```diff
@@ -444,6 +441,24 @@ impl<T: Trait> Module<T> {
         Ok(())
     }
 
+    /// Makes sure that the `secondary_key` has access to the portfolio.
+    fn ensure_portfolio_perms(
+        secondary_key: Option<&SecondaryKey<T::AccountId>>,
+        portfolio: &PortfolioId,
+    ) -> DispatchResult {
+        // Default portfolio are always valid. Custom portfolios must be created explicitly.
+        if let PortfolioKind::User(num) = portfolio.kind {
+            if let Some(sk) = secondary_key {
+                // Check that the secondary signer is allowed to work with this portfolio.
+                ensure!(
+                    sk.has_portfolio_permission(iter::once(num)),
+                    Error::<T>::SecondaryKeyNotAuthorizedForPortfolio
+                );
+            }
+        }
```

Comment:
```suggestion
        if let (PortfolioKind::User(num), Some(sk)) = (portfolio.kind, secondary_key) {
            // Check that the secondary signer is allowed to work with this portfolio.
            ensure!(
                sk.has_portfolio_permission(iter::once(num)),
                Error::<T>::SecondaryKeyNotAuthorizedForPortfolio
            );
        }
```
Or better yet, reuse this `if let Some(...) { ... }` logic inside `ensure_portfolio_validity`.